### PR TITLE
Fixes #23711: Add GPG signature verification to rudder-package

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/23711

Interesting points:

* Finally not using sequoia or another Rust lib as they are too low level and we don't want to deal with gpg internals.
* Switching from `gpg` to [`gpgv`](https://www.gnupg.org/documentation/manuals/gnupg24/gpgv.1.html), simpler, dedicated to signature verification, and provided with gpg.
* Instead of using the public key, directly package a keyring, this makes the runtime check simpler.
* Embed the keyring in the repo. It is not worse than storing a hash of the file and downloading it.

### CODE REVIEW

File: relay/sources/rudder-package/src/signature.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    use ::std::fs::read_to_string;
+    use assert_json_diff::assert_json_eq;
+    use pretty_assertions::assert_eq;
+    use std::str::FromStr;
+    use tempfile::TempDir;
+
+    #[test]
+    fn test_plugin_database_parsing() {
+        let data = read_to_string("./tests/plugin_database_parsing.json")
+            .expect("Unable to parse file './tests/pluginèdatabase_parsing.json'");
```
Comment (Fdall) [inline relay/sources/rudder-package/src/signature.rs:52]:
```suggestion
            .expect("Unable to parse file './tests/plugin_database_parsing.json'");
```

File: relay/sources/rudder-package/src/signature.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+use std::io::{self, Read};
+use crate::SIGNATURE_KEY_PATH;
+
+// This fetches keys and computes the validity of the verification.
+struct Helper {}
+impl VerificationHelper for Helper {
+    fn get_certs(&mut self, _ids: &[KeyHandle]) -> openpgp::Result<Vec<Cert>> {
+
+    // Dependency injection is not possible here, fallback to conditionnal compilation
+    if cfg!(test) {
```
Comment (Fdall) [inline relay/sources/rudder-package/src/signature.rs:18]: I do not understand this part

Comment (amousset) [reply relay/sources/rudder-package/src/signature.rs:18]: this is absolutely not finished, probably not worth reviewing now

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Review by Fdall — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**